database/functions.js

```diff
@@ -215,6 +225,10 @@ const getForms = function(health_zone_id, validation_status, callback, user="")
 // helper function for the drug data dashboard
 const getDrugData = async function(health_zone_id, numPastDays) { 
     try {
+        if (!isValidObjectId(health_zone_id)) {
```

Comment:
This is not enough to validate an object id, "for general primitive strings and even string objects, any string of length 12 passes." I used the code below on Nick's code, and I think it's the way to go.

`if (mongoose.Types.ObjectId.isValid(rawBody._id) &&
        (String)(new mongoose.Types.ObjectId(rawBody._id)) === rawBody._id) {`

Just replace with health zone id. Make this change on all checks you made.

Reply:
Yeah I made sure to do that; I put the validation code in the isValidObjectId function at the top of the functions.js file (lines 12-20)